nixos/modules/services/networking/firefox/sync-server.nix

```diff
@@ -124,7 +124,13 @@ in
       description = "Firefox Sync Server";
       wantedBy = [ "multi-user.target" ];
       path = [ pkgs.pythonPackages.pasteScript pkgs.coreutils ];
-      environment.PYTHONPATH = "${pkgs.pythonPackages.syncserver}/lib/${pkgs.pythonPackages.python.libPrefix}/site-packages";
+      environment.PYTHONPATH = let
+          penv = pkgs.python.buildEnv.override {
+            extraLibs = [ pkgs.pythonPackages.syncserver ];
+            ignoreCollisions = true;
+          };
+        in "${penv}/${pkgs.python.sitePackages}/";
```

Comment:
> Actually syncserver has a two versions of wsgiproxy in its dependencies, which causes a conflict with `python.withPackages`

Ah, so not even considering `paster`, there is already a collision? In that case the conflict isn't with `withPackages`, its simply what it stated: there's two different versions. That means the solution here is brittle and depends on order of `PYTHONPATH` and whether or not packages can deal with a possibly incorrect version of `wsgiproxy`. 

Can you tell which versions of `wsgiproxy`, because on master we have only one version as far as I can tell.

> Well, paster needs to find syncserver somehow. I tested your patch, I get "pkg_resources.DistributionNotFound: The 'syncserver' distribution was not found and is required by the application" from paster

OK, so if `paster` needs to import from `syncserver`, then they need to be in the same env/`PYTHONPATH`.

Your solution can work, as you've experienced, but if there is indeed a collision, then it may easily break.